src/tools/segmentation/BrushTool.js

```diff
@@ -0,0 +1,135 @@
+import external from './../../externalModules.js';
+import { BaseBrushTool } from '../base';
+import store from './../../store/index.js';
+import brushUtils from '../../util/segmentation/brush/index.js';
+import EVENTS from '../../events.js';
+import { getLogger } from '../../util/logger.js';
+
+const logger = getLogger('tools:BrushTool');
+
+const { drawBrushPixels, getCircle } = brushUtils;
+
+const segmentationModule = store.modules.segmentation;
+
+/**
+ * @public
+ * @class BrushTool
+ * @memberof Tools.Brush
+ * @classdesc Tool for drawing segmentations on an image.
+ * @extends Tools.Base.BaseBrushTool
+ */
+export default class BrushTool extends BaseBrushTool {
+  constructor(props = {}) {
+    const defaultProps = {
+      name: 'Brush',
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: { alwaysEraseOnClick: false },
+    };
+
+    super(props, defaultProps);
+
+    this.touchDragCallback = this._paint.bind(this);
+  }
+
+  /**
+   * Called by the event dispatcher to render the image.
+   *
+   * @param {Object} evt - The event.
+   * @returns {void}
+   */
+  renderBrush(evt) {
+    const eventData = evt.detail;
+    const viewport = eventData.viewport;
+
+    let mousePosition;
+
+    if (this._drawing) {
+      mousePosition = this._lastImageCoords;
+    } else if (this._mouseUpRender) {
+      mousePosition = this._lastImageCoords;
+      this._mouseUpRender = false;
+    } else {
+      mousePosition = store.state.mousePositionImage;
+    }
+
+    if (!mousePosition) {
+      return;
+    }
+
+    const { rows, columns } = eventData.image;
+    const { x, y } = mousePosition;
+
+    if (x < 0 || x > columns || y < 0 || y > rows) {
+      return;
+    }
+
+    // Draw the hover overlay on top of the pixel data
+    const radius = segmentationModule.configuration.radius;
+    const context = eventData.canvasContext;
+    const element = eventData.element;
+    const color = segmentationModule.getters.brushColor(element, this._drawing);
+
+    context.setTransform(1, 0, 0, 1, 0, 0);
+
+    const { cornerstone } = external;
+
+    const circleRadius = radius * viewport.scale;
+    const mouseCoordsCanvas = cornerstone.pixelToCanvas(element, mousePosition);
+
+    context.beginPath();
+    context.strokeStyle = color;
+    context.ellipse(
+      mouseCoordsCanvas.x,
+      mouseCoordsCanvas.y,
+      circleRadius,
+      circleRadius,
+      0,
+      0,
+      2 * Math.PI
+    );
```

Comment:
canvas cursor?

Reply:
Yeah, a brush can be way bigger than the 128x128 pixels our svg cursors allow, so until/if we get an SVG layer we are stuck with canvas cursors for now.